webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java

```diff
@@ -3851,6 +3851,49 @@ public String getUserProperty(
     }
 
     /**
+     * Sets a property for the user
+     *
+     * @param sessionid
+     *            The session id.
+     * @param eid
+     *            The user eid.
+     * @param key
+     *            The property key.
+     * @param value
+     *             The property value.
+     * @return
+     *			  Success or exception message
+     */
+    @WebMethod
+    @Path("/setUserProperty")
+    @Produces("text/plain")
+    @GET
+    public String setUserProperty(
+            @WebParam(name = "sessionid", partName = "sessionid") @QueryParam("sessionid") String sessionid,
+            @WebParam(name = "eid", partName = "eid") @QueryParam("eid") String eid,
+            @WebParam(name = "key", partName = "key") @QueryParam("key") String key,
+            @WebParam(name = "value", partName = "value") @QueryParam("value") String value){
+        Session session = establishSession(sessionid);
+
+        if (!securityService.isSuperUser(session.getUserId())) {
+            LOG.warn("WS setUserProperty(): Permission denied. Restricted to super users.");
+            throw new RuntimeException("WS setUserProperty(): Permission denied. Restricted to super users.");
+        }
+
+        try {
+            String userid = userDirectoryService.getUserByEid(eid).getId();
+            UserEdit user = userDirectoryService.editUser(userid);
+            user.getPropertiesEdit().addProperty(key, value);
+            userDirectoryService.commitEdit(user);
+        }
+        catch (Exception e) {
+            LOG.error("WS setUserProperty(): " + e.getClass().getName() + " : " + e.getMessage(), e);
+            return e.getClass().getName() + " : " + e.getMessage();
```

Comment:
I think it would be better to do one of the following:
1) just return "failure" and log the error typically as a warning (best to do this when there is no exception, or when an exception is not appropriate)
2) let the checked exception bubble (bad, requires more work as exceptions needs to be declared)
3) wrap the checked exception in a RuntimeException (better)
4) create single service exception that can be used everywhere (better, and isolates exceptions from RuntimeExceptions)

So in Sakai you will find the use of 1 and 3, it probably would be nice to have 4 but we don't have it!

Long story short I would change 3891 to return "failure" and change LOG.error to a LOG.warn!